src/main/scala/tech/cryptonomic/conseil/Lorre.scala

```diff
@@ -20,14 +21,15 @@ object Lorre extends App with LazyLogging {
   private val sleepIntervalInSeconds = conf.getInt("lorre.sleepIntervalInSeconds")
   private val feeUpdateInterval = conf.getInt("lorre.feeUpdateInterval")
   private val purgeAccountsInterval = conf.getInt("lorre.purgeAccountsInterval")
+  private val network = conf.getString("lorre.tezos-network")
 
   lazy val db = DatabaseUtil.db
   val tezosNodeOperator = new TezosNodeOperator(TezosNodeInterface)
 
   var iterationsOfLorre = 0
 
-  try {
-    while(true) {
+  @tailrec
```

Comment:
Wait, you need an annotation to make functions tail recursive? Would this speed up efficiency considerably?

Reply:
The annotation is only a compile-time check that will fail the compilation if the method is not actually tail-recursive.

Even if you don't add the annotation, `scalac` will rewrite the function optimizing the tail-call when possible. This leaves open the risk that you have incorrectly written the method, assuming it's tail-recursive when it's not, so the annotation is an hint that you expect it to be so, and the compiler confirms it.

In this case the code is pretty obvious, somehow, but it's also critical to check because a loop like this would soon exhaust the stack if not tail-rec